crystal-mode.el

```diff
@@ -2757,5 +2767,6 @@ directory of the current file."
   (add-to-list 'interpreter-mode-alist (cons (purecopy name) 'crystal-mode)))
 
 (provide 'crystal-mode)
+(provide 'crystal-tool--project-name)
```

Comment:
Function export?

Reply:
The plan was to not have to have the same function defined in both this and the flycheck file. Perhaps there is a better way to do it?

Reply:
@yxhuvud *provide* and *require* are an alternative to autoload for loading files automatically. https://www.gnu.org/software/emacs/manual/html_node/elisp/Named-Features.html

Reply:
Ok, I had some issues before but they don't seem to happen anymore so I removed it and the require statement as the latter seems to break byte compilation for me.